plutus-contract/src/Plutus/Contract/Test/ContractModel/Internal.hs

```diff
@@ -1579,30 +1580,34 @@ checkNoLockedFundsProofWithWiggleRoomFast wiggle = checkNoLockedFundsProofWithWi
 checkNoLockedFundsProofWithWiggleRoom'
   :: (ContractModel model)
   => Integer
-  -> (Actions model -> Property)
+  -> (StateModel.Actions (ModelState model) -> Property)
   -> NoLockedFundsProof model
   -> Property
 checkNoLockedFundsProofWithWiggleRoom' wiggle run NoLockedFundsProof{nlfpMainStrategy   = mainStrat,
                                                                      nlfpWalletStrategy = walletStrat } =
     forAllDL anyActions_ $ \ (Actions as) ->
     forAllUniqueDL (nextVarIdx as) (stateAfter $ Actions as) mainStrat $ \ (Actions as') ->
           let s = stateAfter $ Actions (as ++ as') in
-            foldl (QC..&&.) (run (Actions $ as ++ as') QC..&&. (counterexample "Main strategy" . counterexample (show . Actions $ as ++ as') $ prop s))
```

Comment:
Use the new `Semigroup Action` instance here in the next refactor.

Reply:
A different kind of actions ;)

Reply:
But we should add a semigroup instance for those actions too...